plugins/out_datadog/datadog.c

```diff
@@ -179,7 +179,7 @@ static int datadog_format(struct flb_config *config,
                     return -1;
                 }
             } else if (flb_sds_len(remapped_tags) < byte_cnt) {
-                tmp = flb_sds_increase(remapped_tags, flb_sds_len(remapped_tags) - byte_cnt);
+                tmp = flb_sds_increase(remapped_tags, byte_cnt - flb_sds_len(remapped_tags));
```

Comment:
So it appears that this line of code gets run when you have a short remapped_tags buffer size estimation, followed by a long remapped_tags buffer size estimation.

The remapped_tags buffer is reused for each log in the batch and only resized if we estimate more space is needed.

```
byte_cnt
```
Is used for the buffer size estimation

```
flb_sds_len(remapped_tags)
```
is the current buffer size.

flb_sds_len(remapped_tags) - byte_cnt is going to be negative because we only resize if byte_cnt is greater than the current buffer size. This PR resolves the logic here

Reply:
Thanks Matt! 👏

Reply:
> when you have a short remapped_tags buffer size estimation, followed by a long remapped_tags buffer size estimation.

I think basically since this is a buffer re-used in a loop over all records, that this case is where may be the first log in a flush is missing metadata, so there's no remapping and the size allocated is small. Then next log record has the metadata, so we need to resize. Do you think I'm understanding this code correctly?

Reply:
That is correct, or if the metadata varies in size requiring more buffer space, then the resize would be needed as well.